### PR TITLE
Erez Tal

### PR DESCRIPTION
**Pull request by etal2**
### Short Description
Completed the assignment, my findings can be found in the comments.md file. 

### Issue Reference
fixes #

### Checks

- [ ] This PR follows the coding guidelines of the project
- [ ] This PR fulfills the acceptance criteria defined on the issue
- [ ] This PR has been locally tested
- [ ] Changes in this PR are unit tested

### CODE REVIEW

File: src/comments.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# 3AP Political and Postal communities in Switzerland
+
+## Assignment 2
+- The datasets are correlated using the political community number which appeares in both csv files
+- Postal communities are unique using a composite key using zipCode and zipCodeAdditions together
```
Comment (robertblust) [inline src/comments.md:5]: 👍

File: src/comments.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
+# 3AP Political and Postal communities in Switzerland
+
+## Assignment 2
+- The datasets are correlated using the political community number which appeares in both csv files
```
Comment (robertblust) [inline src/comments.md:4]: 👍

File: src/comments.md [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+- Postal communities are unique using a composite key using zipCode and zipCodeAdditions together
+- I added looking up entities directly by keys
```
Comment (robertblust) [inline src/comments.md:6]: This is inspired I assume from relational databases (foreign keys). On a model side I prefer to get the actual object and not just the key. But this is opinionated 😄

File: src/comments.md [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+- Since the data is pretty much immutable I de-normalize the data and enabled bidirectional relations on some entities
```
Comment (robertblust) [inline src/comments.md:7]: very nice 👍

File: src/comments.md [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+
+## Assignment 3
+- The method getLastUpdateOfPoliticalCommunityByPostalCommunityName has some unclear edge cases. There can be multiple postal communities with the same name and there can be multiple political communities belonging to each postal community. In the case of multiple answers I'm currently picking the latest LastUpdate out of them all but it might be better to just throw on any ambiguity.There are no tests for these edge cases.
```
Comment (robertblust) [inline src/comments.md:10]: Excellent catch. This test cases needs to be changed. Does not makes sense 🥇

**Review by robertblust — COMMENTED** [review]
@etal2 very nice solution

* I really like the well described findings in https://github.com/3AP-AG/candidate-assignment-java/blob/b84dd6df40531b80f07b35661a5febe8bb3e88fb/src/comments.md
* Good usage of lombok
* ModelFactory approach 👍 

You will be contacted by Simona latest next Week.

Cheers @robertblust
